### PR TITLE
ci, build: update CI, libraries, python dependencies, and format

### LINKED ISSUE #274: `jupyter-ui` unit test fails with a SDI error
It seems like the latest SDI version is introducing issues to the unit tests and the charm (potentially) as one of the test cases of this charm is failing with the following message:

```
serialized_data_interface.errors.MissingSchemaError: No provides schema found for data on ingress:0 from istio-pilot
```

For logs and more details, go to the latest [unit test CI run](https://github.com/canonical/notebook-operators/actions/runs/5643000507/job/15283969379#step:5:101) of #273

### PR DESCRIPTION
**Pull request by DnPlas**
This commit updates the CI with standard CI configuration (juju, actions, deps) that is shared across multiple repositories. All python dependencies are also updated to their latest working versions. This PR also includes library updates. Finally, some formatting was applied on non-compliant files (copyright, style).

Fixes #274, #275

### CODE REVIEW

File: requirements-integration.in [hunk 1]
```diff
@@ -1,4 +1,5 @@
-juju<3.1
+# Pinning to <3.0 to ensure compatibility with the 2.9 controller version
+juju<3.0
 lightkube
 pytest-operator
```
Comment (phoevos) [inline requirements-integration.in:4]: Redundant.

File: requirements-integration.in
Comment (phoevos) [inline requirements-integration.in:1]: Let's please fix ordering.

**Review by phoevos — CHANGES_REQUESTED** [review]
```diff
diff --git a/charms/jupyter-controller/requirements-lint.in b/charms/jupyter-controller/requirements-lint.in
index c77cd87..07a4a51 100644
--- a/charms/jupyter-controller/requirements-lint.in
+++ b/charms/jupyter-controller/requirements-lint.in
@@ -1,9 +1,8 @@
+black
 codespell
-flake8<6
+flake8
 flake8-builtins
-# Pinned because `flake8-copyright==0.2.3` is incompatible with `flake8>=6`.  Can unpin this
-# when https://github.com/savoirfairelinux/flake8-copyright/pull/20 or a similar fix is released
 flake8-copyright
+isort
 pep8-naming
 pyproject-flake8
--r requirements-fmt.txt
diff --git a/charms/jupyter-controller/requirements.in b/charms/jupyter-controller/requirements.in
index c5fae6a..44f023c 100644
--- a/charms/jupyter-controller/requirements.in
+++ b/charms/jupyter-controller/requirements.in
@@ -1,4 +1,3 @@
 charmed-kubeflow-chisme
 lightkube
-lightkube-models
 ops
```

File: requirements-integration.in (hunk 1 above)
Comment (DnPlas) [reply requirements-integration.in:4]: What do you mean?

Comment (phoevos) [reply requirements-integration.in]: It's not imported directly and is probably only needed for the asyncio stuff we do on the pytest invocation. You can leave it in I guess.

**Review by phoevos — APPROVED** [review]

**Pull request merged**